internal/service/lexv2models/slot.go

```diff
@@ -0,0 +1,894 @@
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.LexV2Models, create.ErrActionDeleting, ResNameSlot, state.ID.String(), err),
+			err.Error(),
+		)
+		return
+	}
+}
+
+func findSlotByID(ctx context.Context, conn *lexmodelsv2.Client, id string) (*lexmodelsv2.DescribeSlotOutput, error) {
+	parts, err := intflex.ExpandResourceId(id, slotIDPartCount, false)
+	if err != nil {
+		return nil, err
+	}
+
+	in := &lexmodelsv2.DescribeSlotInput{
+		BotId:      aws.String(parts[0]),
+		BotVersion: aws.String(parts[1]),
+		IntentId:   aws.String(parts[2]),
+		LocaleId:   aws.String(parts[3]),
+		SlotId:     aws.String(parts[4]),
+	}
+
+	out, err := conn.DescribeSlot(ctx, in)
+
+	if errs.IsA[*awstypes.ResourceNotFoundException](err) {
+		return nil, &retry.NotFoundError{
+			LastError:   err,
+			LastRequest: in,
+		}
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	if out == nil {
+		return nil, tfresource.NewEmptyResultError(in)
+	}
+
+	return out, nil
+}
+
+type resourceSlotData struct {
+	BotID                   types.String                                                 `tfsdk:"bot_id"`
+	BotVersion              types.String                                                 `tfsdk:"bot_version"`
+	Description             types.String                                                 `tfsdk:"description"`
+	ID                      types.String                                                 `tfsdk:"id"`
+	IntentID                types.String                                                 `tfsdk:"intent_id"`
+	LocaleID                types.String                                                 `tfsdk:"locale_id"`
+	SlotID                  types.String                                                 `tfsdk:"slot_id"`
+	MultipleValuesSetting   fwtypes.ListNestedObjectValueOf[MultipleValuesSettingData]   `tfsdk:"multiple_values_setting"`
+	Name                    types.String                                                 `tfsdk:"name"`
+	ObfuscationSetting      fwtypes.ListNestedObjectValueOf[ObfuscationSettingData]      `tfsdk:"obfuscation_setting"`
+	Timeouts                timeouts.Value                                               `tfsdk:"timeouts"`
+	SlotTypeID              types.String                                                 `tfsdk:"slot_type_id"`
+	ValueElicitationSetting fwtypes.ListNestedObjectValueOf[ValueElicitationSettingData] `tfsdk:"value_elicitation_setting"`
+}
+
+type MultipleValuesSettingData struct {
+	AllowMultipleValues types.Bool `tfsdk:"allow_multiple_values"`
+}
+
+type ObfuscationSettingData struct {
+	ObfuscationSettingType fwtypes.StringEnum[awstypes.ObfuscationSettingType] `tfsdk:"obfuscation_setting_type"`
+}
+
+type DefaultValueSpecificationData struct {
+	DefaultValueList fwtypes.ListNestedObjectValueOf[DefaultValueData] `tfsdk:"default_value_list"`
+}
+
+type DefaultValueData struct {
+	DefaultValue types.String `tfsdk:"default_value"`
+}
+
+type PromptSpecificationData struct {
```

Comment:
Using the type in intent unless it's different. Should be relatively drop-in replace. Remove `PromptSpecificationData` type declaration and use `PromptSpecification` whereever you find `PromptSpecificationData`.